src/libproc_macro/lib.rs

```diff
@@ -348,6 +348,46 @@ impl Span {
         }
     }
 
+    /// Produces a span pointing to the first byte of this span.
+    ///
+    /// When used on the span of a [`Group`], the first byte is the opening
+    /// delimiter of the group. If we have a span pointing to the `(self)`
+    /// argument list in the line below, the span produced by `at_start()` would
+    /// be:
+    ///
+    /// ```text
+    /// pub fn at_start(self) -> Span {
+    ///                ^
+    /// ```
+    #[stable(feature = "proc_macro_span_start_end", since = "1.30.0")]
```

Comment:
@parched 
Yeah, `len()` + `Index<RangeBounds<usize>>` for spans, basically - `span[a .. b]`, `span[a..]`.
Except not `Index` exactly, because it must return a reference.

That would be a small, self-contained, but powerful API.
I like it.